File: 729-my-calendar-i/729-my-calendar-i.cs
public class MyCalendar {
    List<int[]> ls= null;
    public MyCalendar() {
        ls=new List<int[]>();
    }
    
    public bool Book(int start, int end) {
        foreach(var item in ls){
            if(end>item[0] && start<item[1])
                return false;
        }
        ls.Add(new int[]{start,end});
        return true;
    }
}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar obj = new MyCalendar();
 * bool param_1 = obj.Book(start,end);
 */

File: 729-my-calendar-i/729-my-calendar-i.cpp
class MyCalendar {
public:
    set<pair<int,int>> m;
    MyCalendar() {
        
    }
    
    bool book(int s, int e) {
        auto next = m.lower_bound({s,e}); // first element with key not go before k (i.e., either it is equivalent or goes after).
        
        // a existing booking started within the new booking (next)
        if(next != m.end() && next->first < e) return false;
        
        // new booking started within a existing booking (prev)
        if(next != m.begin() && s < (--next)->second) return false;
        
        m.insert({s,e});
            
        return true;
    }
};

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar* obj = new MyCalendar();
 * bool param_1 = obj->book(start,end);
 */